cc-snapshot

```diff
@@ -117,6 +118,18 @@ if [ "$(version "$openstackclient_version")" -lt "$(version "3.16.0")" ]; then
 fi
 ###### Check openstackclient version end ######
 
+###### Get provenance properties ######
+PROVENANCE_PROPERTIES=''
+if [ -f "$PROVENANCE_FILE" ]; then
+  for prop in $(cat $PROVENANCE_FILE | tr -d '\n' | tr -d [:space:] | tr -d '{' | tr -d '}' | tr ',' '\n');
+  do
+    prop="${prop/\":\"/=}"
+    prop="$(echo $prop | tr -d '"')"
+    PROVENANCE_PROPERTIES="--property provenance-$prop $PROVENANCE_PROPERTIES"
```

Comment:
FYI another way of doing this is to use a shell array

```shell
declare -a PROVENANCE_PROPERTIES=()
PROVENANCE_PROPERTIES+=(--property provenance-$prop)
openstack image create ${PROVENANCE_PROPERTIES[@]} ...
```

Reply:
It's good to know! But seems like the bash version of our images couldn't handle empty shell array, and need extra `if` statement. `${PROVENANCE_PROPERTIES[@]}` gives `unbound variable` error if empty.